diff --git a/index.esm.js b/index.esm.js
index 87bf99204d5c4342f153b80128e0ce3b0d7efdbc..8fef7215bce78e475b042e236941e2a8ed783ab5 100644
--- a/index.esm.js
+++ b/index.esm.js
@@ -444,11 +444,8 @@
           throw HttpZError$5.get('Incorrect header row format, expected: Name: Value', hRow)
         }
 
-        // quoted string must be parsed as a single value (https://tools.ietf.org/html/rfc7230#section-3.2.6)
         if (_$6.isNil(value)) {
           value = '';
-        } else if (consts$6.regexps.quoutedHeaderValue.test(value)) {
-          value = _$6.trim(value, '"');
         }
 
         return {
diff --git a/src/parsers/base.js b/src/parsers/base.js
index 25a141697d3f27324fe1fd4f7ed7d30832d417e0..33475cb41f952775d99fb4b7ec82732ed956204f 100644
--- a/src/parsers/base.js
+++ b/src/parsers/base.js
@@ -34,11 +34,8 @@ class HttpZBaseParser {
         throw HttpZError.get('Incorrect header row format, expected: Name: Value', hRow)
       }
 
-      // quoted string must be parsed as a single value (https://tools.ietf.org/html/rfc7230#section-3.2.6)
       if (_.isNil(value)) {
         value = ''
-      } else if (consts.regexps.quoutedHeaderValue.test(value)) {
-        value = _.trim(value, '"')
       }
 
       return {
