diff --git a/src/bash-parser-web.ts b/src/bash-parser-web.ts
index 2cc594f..b97e4df 100644
--- a/src/bash-parser-web.ts
+++ b/src/bash-parser-web.ts
@@ -1,11 +1,29 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
 import Parser from "web-tree-sitter";
+// @ts-ignore
+import wasmParser from "../../../web-tree-sitter/tree-sitter.wasm?url";
+// @ts-ignore
+import wasmBash from "../tree-sitter-bash.wasm?url";
+
+export default new Promise<Parser>(async (resolve, reject) => {
+  try {
+    await Parser.init({
+      locateFile: () => {
+        if (typeof wasmParser === 'string') {
+          return wasmParser;
+        }
 
-// NOTE: Top-level await is not available in Safari until Safari 15
-// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#browser_compatibility
-await Parser.init();
-const Bash = await Parser.Language.load("/tree-sitter-bash.wasm");
-const parser = new Parser();
-parser.setLanguage(Bash);
+        return URL.createObjectURL(new Blob([wasmParser]));
+      }
+    });
+    const Bash = await Parser.Language.load(wasmBash);
+    const parser = new Parser();
+    parser.setLanguage(Bash);
 
-export default parser;
+    resolve(parser);
+  } catch (err) {
+    reject(err);
+  }
+});
 export type { Parser };
diff --git a/src/generators/ansible.ts b/src/generators/ansible.ts
index d0f2e29..3a1eb3a 100644
--- a/src/generators/ansible.ts
+++ b/src/generators/ansible.ts
@@ -111,14 +111,14 @@ export const _toAnsible = (
     2
   );
 };
-export const toAnsibleWarn = (
+export const toAnsibleWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const ansible = _toAnsible(request, warnings);
   return [ansible, warnings];
 };
-export const toAnsible = (curlCommand: string | string[]): string => {
-  return toAnsibleWarn(curlCommand)[0];
+export const toAnsible = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toAnsibleWarn(curlCommand))[0];
 };
diff --git a/src/generators/cfml.ts b/src/generators/cfml.ts
index 52243e2..919f4dc 100644
--- a/src/generators/cfml.ts
+++ b/src/generators/cfml.ts
@@ -136,15 +136,15 @@ export const _toCFML = (request: Request, warnings: Warnings = []): string => {
   return cfmlCode;
 };
 
-export const toCFMLWarn = (
+export const toCFMLWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const cfml = _toCFML(request, warnings);
   return [cfml, warnings];
 };
 
-export const toCFML = (curlCommand: string | string[]): string => {
-  return toCFMLWarn(curlCommand)[0];
+export const toCFML = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toCFMLWarn(curlCommand))[0];
 };
diff --git a/src/generators/dart.ts b/src/generators/dart.ts
index 9c370b3..0e719c1 100644
--- a/src/generators/dart.ts
+++ b/src/generators/dart.ts
@@ -142,14 +142,14 @@ export const _toDart = (request: Request, warnings: Warnings = []): string => {
 
   return s + "\n";
 };
-export const toDartWarn = (
+export const toDartWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const dart = _toDart(request, warnings);
   return [dart, warnings];
 };
-export const toDart = (curlCommand: string | string[]): string => {
-  return toDartWarn(curlCommand)[0];
+export const toDart = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toDartWarn(curlCommand))[0];
 };
diff --git a/src/generators/elixir.ts b/src/generators/elixir.ts
index 7d300f5..da7ad10 100644
--- a/src/generators/elixir.ts
+++ b/src/generators/elixir.ts
@@ -249,15 +249,15 @@ response = HTTPoison.request(request)
 
   return template;
 };
-export const toElixirWarn = (
+export const toElixirWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const elixir = _toElixir(request, warnings);
   return [elixir, warnings];
 };
 
-export const toElixir = (curlCommand: string | string[]): string => {
-  return toElixirWarn(curlCommand)[0];
+export const toElixir = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toElixirWarn(curlCommand))[0];
 };
diff --git a/src/generators/go.ts b/src/generators/go.ts
index 9626460..8a14058 100644
--- a/src/generators/go.ts
+++ b/src/generators/go.ts
@@ -93,14 +93,14 @@ export const _toGo = (request: Request, warnings: Warnings = []): string => {
 
   return goCode + "\n";
 };
-export const toGoWarn = (
+export const toGoWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const go = _toGo(request, warnings);
   return [go, warnings];
 };
-export const toGo = (curlCommand: string | string[]): string => {
-  return toGoWarn(curlCommand)[0];
+export const toGo = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toGoWarn(curlCommand))[0];
 };
diff --git a/src/generators/java.ts b/src/generators/java.ts
index 53f4d84..95c5f63 100644
--- a/src/generators/java.ts
+++ b/src/generators/java.ts
@@ -114,15 +114,15 @@ export const _toJava = (request: Request, warnings: Warnings = []): string => {
 
   return javaCode + "\n";
 };
-export const toJavaWarn = (
+export const toJavaWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const java = _toJava(request, warnings);
   return [java, warnings];
 };
 
-export const toJava = (curlCommand: string | string[]): string => {
-  return toJavaWarn(curlCommand)[0];
+export const toJava = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toJavaWarn(curlCommand))[0];
 };
diff --git a/src/generators/javascript/axios.ts b/src/generators/javascript/axios.ts
index c3e8a4e..8c0691b 100644
--- a/src/generators/javascript/axios.ts
+++ b/src/generators/javascript/axios.ts
@@ -377,14 +377,14 @@ export const _toNodeAxios = (
 
   return importCode + "\n" + code;
 };
-export const toNodeAxiosWarn = (
+export const toNodeAxiosWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const nodeAxios = _toNodeAxios(request, warnings);
   return [nodeAxios, warnings];
 };
-export const toNodeAxios = (curlCommand: string | string[]): string => {
-  return toNodeAxiosWarn(curlCommand)[0];
+export const toNodeAxios = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toNodeAxiosWarn(curlCommand))[0];
 };
diff --git a/src/generators/javascript/javascript.ts b/src/generators/javascript/javascript.ts
index fac85da..6b6d74d 100644
--- a/src/generators/javascript/javascript.ts
+++ b/src/generators/javascript/javascript.ts
@@ -320,32 +320,32 @@ export const _toNode = (request: Request, warnings: Warnings = []): string => {
   return _toJavaScriptOrNode(request, warnings, true);
 };
 
-export const toJavaScriptWarn = (
+export const toJavaScriptWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(
     curlCommand,
     javaScriptSupportedArgs,
     warnings
   );
   return [_toJavaScript(request, warnings), warnings];
 };
-export const toJavaScript = (curlCommand: string | string[]): string => {
-  return toJavaScriptWarn(curlCommand)[0];
+export const toJavaScript = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toJavaScriptWarn(curlCommand))[0];
 };
 
-export const toNodeWarn = (
+export const toNodeWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(
     curlCommand,
     nodeSupportedArgs,
     warnings
   );
   return [_toNode(request, warnings), warnings];
 };
-export const toNode = (curlCommand: string | string[]): string => {
-  return toNodeWarn(curlCommand)[0];
+export const toNode = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toNodeWarn(curlCommand))[0];
 };
diff --git a/src/generators/javascript/node-request.ts b/src/generators/javascript/node-request.ts
index 912d326..57fb11b 100644
--- a/src/generators/javascript/node-request.ts
+++ b/src/generators/javascript/node-request.ts
@@ -99,16 +99,16 @@ export const _toNodeRequest = (
 
   return nodeRequestCode + "\n";
 };
-export const toNodeRequestWarn = (
+export const toNodeRequestWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   warnings.unshift(["node-request", "the request package is deprecated"]);
 
   const nodeRequests = _toNodeRequest(request, warnings);
   return [nodeRequests, warnings];
 };
-export const toNodeRequest = (curlCommand: string | string[]): string => {
-  return toNodeRequestWarn(curlCommand)[0];
+export const toNodeRequest = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toNodeRequestWarn(curlCommand))[0];
 };
diff --git a/src/generators/json.ts b/src/generators/json.ts
index f25f862..b6c6a10 100644
--- a/src/generators/json.ts
+++ b/src/generators/json.ts
@@ -183,14 +183,14 @@ export const _toJsonString = (
     ) + "\n"
   );
 };
-export const toJsonStringWarn = (
+export const toJsonStringWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const json = _toJsonString(request, warnings);
   return [json, warnings];
 };
-export const toJsonString = (curlCommand: string | string[]): string => {
-  return toJsonStringWarn(curlCommand)[0];
+export const toJsonString = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toJsonStringWarn(curlCommand))[0];
 };
diff --git a/src/generators/matlab/matlab.ts b/src/generators/matlab/matlab.ts
index a66776f..a487e7d 100644
--- a/src/generators/matlab/matlab.ts
+++ b/src/generators/matlab/matlab.ts
@@ -40,14 +40,14 @@ export const _toMATLAB = (
     .filter((line) => line !== null)
     .join("\n");
 };
-export const toMATLABWarn = (
+export const toMATLABWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const matlab = _toMATLAB(request, warnings);
   return [matlab, warnings];
 };
-export const toMATLAB = (curlCommand: string | string[]): string => {
-  return toMATLABWarn(curlCommand)[0];
+export const toMATLAB = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toMATLABWarn(curlCommand))[0];
 };
diff --git a/src/generators/php/php-requests.ts b/src/generators/php/php-requests.ts
index 77f649d..a469fd5 100644
--- a/src/generators/php/php-requests.ts
+++ b/src/generators/php/php-requests.ts
@@ -114,14 +114,14 @@ export const _toPhpRequests = (
 
   return phpCode + "\n";
 };
-export const toPhpRequestsWarn = (
+export const toPhpRequestsWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const php = _toPhpRequests(request, warnings);
   return [php, warnings];
 };
-export const toPhpRequests = (curlCommand: string | string[]): string => {
-  return toPhpRequestsWarn(curlCommand)[0];
+export const toPhpRequests = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toPhpRequestsWarn(curlCommand))[0];
 };
diff --git a/src/generators/php/php.ts b/src/generators/php/php.ts
index 160b958..058a6b3 100644
--- a/src/generators/php/php.ts
+++ b/src/generators/php/php.ts
@@ -156,14 +156,14 @@ export const _toPhp = (request: Request, warnings: Warnings = []): string => {
   return phpCode;
 };
 
-export const toPhpWarn = (
+export const toPhpWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const php = _toPhp(request, warnings);
   return [php, warnings];
 };
-export const toPhp = (curlCommand: string | string[]): string => {
-  return toPhpWarn(curlCommand)[0];
+export const toPhp = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toPhpWarn(curlCommand))[0];
 };
diff --git a/src/generators/python.ts b/src/generators/python.ts
index cd1b713..fc15714 100644
--- a/src/generators/python.ts
+++ b/src/generators/python.ts
@@ -1026,15 +1026,15 @@ export const _toPython = (
   return pythonCode + "\n";
 };
 
-export const toPythonWarn = (
+export const toPythonWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const python = _toPython(request, warnings);
   return [python, warnings];
 };
 
-export const toPython = (curlCommand: string | string[]): string => {
-  return toPythonWarn(curlCommand)[0];
+export const toPython = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toPythonWarn(curlCommand))[0];
 };
diff --git a/src/generators/r.ts b/src/generators/r.ts
index 8ce5305..7601028 100644
--- a/src/generators/r.ts
+++ b/src/generators/r.ts
@@ -221,14 +221,14 @@ export const _toR = (request: Request, warnings: Warnings = []): string => {
 
   return rstatsCode + "\n";
 };
-export const toRWarn = (
+export const toRWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const r = _toR(request, warnings);
   return [r, warnings];
 };
-export const toR = (curlCommand: string | string[]): string => {
-  return toRWarn(curlCommand)[0];
+export const toR = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toRWarn(curlCommand))[0];
 };
diff --git a/src/generators/rust.ts b/src/generators/rust.ts
index 8403be7..883e08c 100644
--- a/src/generators/rust.ts
+++ b/src/generators/rust.ts
@@ -135,14 +135,14 @@ export const _toRust = (request: Request, warnings: Warnings = []): string => {
 
   return lines.join("\n") + "\n";
 };
-export const toRustWarn = (
+export const toRustWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const rust = _toRust(request, warnings);
   return [rust, warnings];
 };
-export const toRust = (curlCommand: string | string[]): string => {
-  return toRustWarn(curlCommand)[0];
+export const toRust = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toRustWarn(curlCommand))[0];
 };
diff --git a/src/generators/strest.ts b/src/generators/strest.ts
index 3089f68..9340684 100644
--- a/src/generators/strest.ts
+++ b/src/generators/strest.ts
@@ -140,14 +140,14 @@ export const _toStrest = (
 
   return yaml.stringify(response, 100, 2);
 };
-export const toStrestWarn = (
+export const toStrestWarn = async (
   curlCommand: string | string[],
   warnings: Warnings = []
-): [string, Warnings] => {
-  const request = util.parseCurlCommand(curlCommand, supportedArgs, warnings);
+): Promise<[string, Warnings]> => {
+  const request = await util.parseCurlCommand(curlCommand, supportedArgs, warnings);
   const strest = _toStrest(request, warnings);
   return [strest, warnings];
 };
-export const toStrest = (curlCommand: string | string[]): string => {
-  return toStrestWarn(curlCommand)[0];
+export const toStrest = async (curlCommand: string | string[]): Promise<string> => {
+  return (await toStrestWarn(curlCommand))[0];
 };
diff --git a/src/util.ts b/src/util.ts
index 140075c..5748af0 100644
--- a/src/util.ts
+++ b/src/util.ts
@@ -1,7 +1,7 @@
 import URL from "url";
 
-import parser from "./bash-parser.js";
-import type { Parser } from "./bash-parser.js";
+import parserPromise from "./bash-parser-web.js";
+import type { Parser } from "./bash-parser-web.js";
 
 // TODO: this type doesn't work.
 function has<T, K extends PropertyKey>(
@@ -985,10 +985,11 @@ interface TokenizeResult {
   stdin?: string;
   stdinFile?: string;
 }
-const tokenize = (
+const tokenize = async (
   curlCommand: string,
   warnings: Warnings = []
-): TokenizeResult => {
+): Promise<TokenizeResult> => {
+  const parser = await parserPromise;
   const ast = parser.parse(curlCommand);
   // https://github.com/tree-sitter/tree-sitter-bash/blob/master/grammar.js
   // The AST must be in a nice format, i.e.
@@ -1789,11 +1790,11 @@ function buildRequest(
   return request;
 }
 
-function parseCurlCommand(
+async function parseCurlCommand(
   curlCommand: string | string[],
   supportedArgs?: Set<string>,
   warnings: Warnings = []
-): Request {
+): Promise<Request> {
   let cmdName: string,
     args: string[],
     stdin: undefined | string,
@@ -1804,7 +1805,7 @@ function parseCurlCommand(
       throw new CCError("no arguments provided");
     }
   } else {
-    ({ cmdName, args, stdin, stdinFile } = tokenize(curlCommand, warnings));
+    ({ cmdName, args, stdin, stdinFile } = await tokenize(curlCommand, warnings));
     if (typeof cmdName === "undefined") {
       throw new CCError("failed to parse input");
     }
diff --git a/tsconfig.json b/tsconfig.json
index 7879258..e298c6b 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -6,7 +6,7 @@
     "sourceMap": true,
     "strict": true,
     "target": "es2018",
-    "lib": ["es2018"],
+    "lib": ["es2018", "dom"],
     "moduleResolution": "node",
     "module": "es2022",
     "esModuleInterop": true,
